client/context.go

```diff
@@ -365,7 +374,7 @@ func GetFromFields(kr keyring.Keyring, from string, genOnly bool) (sdk.AccAddres
 
 // NewKeyringFromBackend gets a Keyring object from a backend
 func NewKeyringFromBackend(ctx Context, backend string) (keyring.Keyring, error) {
-	if ctx.GenerateOnly || ctx.Simulate {
+	if ctx.Simulate {
```

Comment:
This is removed because `--generate-only` should be able to read the key ring (for this feature #9838) and it should not be overwritten.
Moreover this was actually never true as `NewKeyringFromBackend` was called before that the generate flag was read by `readTxCommandFlags`:

- https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/client/cmd.go#L127
- https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/client/cmd.go#L188
- https://github.com/cosmos/cosmos-sdk/blob/v0.46.0-beta2/client/cmd.go#L193